src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRunSymbolNodeFactory.cs

```diff
@@ -18,6 +20,14 @@ public sealed class ReadyToRunSymbolNodeFactory
         public ReadyToRunSymbolNodeFactory(ReadyToRunCodegenNodeFactory codegenNodeFactory)
         {
             _codegenNodeFactory = codegenNodeFactory;
+
+            _genericLookupTypeHelpers = new Dictionary<GenericLookupTypeKey, ISymbolNode>[(int)CORINFO_RUNTIME_LOOKUP_KIND.CORINFO_LOOKUP_KIND_COUNT];
```

Comment:
Nit: I would just make the lookup kind part of the `GenericLookupTypeKey` so that we don't need to have an array of dictionaries (just a single dictionary). It would probably simplify things.